.github/workflows/tests-jellyfish.yml

```diff
@@ -11,12 +11,42 @@ concurrency:
   cancel-in-progress: ${{ github.ref != 'refs/heads/master' }}
 
 env:
-  BUILD_VERSION: latest # Computed
+  DOCKER_HUB_USER: defi
+  DOCKER_IMAGE: defi/defichain:jellyfish-ci-${GITHUB_SHA:0:10}
 
 jobs:
+  docker-build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Populate environment
+        run: ./make.sh ci-export-vars
+
+      - name: Build docker image
+        run: docker build -t ${{ env.DOCKER_IMAGE }} -f ./contrib/dockerfiles/x86_64-pc-linux-gnu.dockerfile .
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ env.DOCKER_HUB_USER }}
+          password: ${{ secrets.DOCKER_HUB_TOKEN }}
+
+      - name: Push to Docker Hub
+        run: docker push ${{ env.DOCKER_IMAGE }}
+
```

Comment:
```suggestion
  build:
    runs-on: ubuntu-latest
    container: defi/ain-builder:latest
    env:
      CARGO_INCREMENTAL: 0

    steps:
    - uses: actions/checkout@v4
    - run: git config --global --add safe.directory '*'

    - name: Populate environment
      run: ./make.sh ci-export-vars

    - name: Setup dependencies
      run: ./make.sh ci-setup-deps

    - name: Setup user dependencies
      run: ./make.sh ci-setup-user-deps

    - name: Restore cpp build cache
      id: cpp-cache-restore
      uses: actions/cache/restore@v3
      with:
        path: |
          ./build/depends
          ./build/src
          ~/.ccache
        key: cpp-${{ env.TARGET }}-${{ env.BUILD_TYPE }}

    - name: Rust build cache
      uses: Swatinem/rust-cache@v2
      id: rust-cache-restore
      with:
        workspaces: lib -> ../build/lib/target
        save-if: ${{ github.ref == 'refs/heads/master' }}
        shared-key: ${{ env.TARGET }}

    - name: Build binaries
      run: ./make.sh build

    - name: Upload binaries
      uses: actions/upload-artifact@v3
      with:
        name: defi-bins
        path: |
          build/src/defid
          build/src/defi-cli
```

Reply:
There shouldnt be a need to push images to the registry since the test workflow is local. In addition, using docker build for this workflow will not be ideal since we cant leverage on the incremental builds on our cpp and rust side, as we cache already our builds to speed up our workflows. Using local builds for the jellyfish test will speed up the CI workflow even more

Reply:
If jellyfishsdk is dependent on defid to be running in a container, then perhaps we can use defi.dockerfile to build our defid image instead. We can re-use the same pipeline to build the image, then upload the built image as an artifact.
 Ref: https://github.com/DeFiCh/ain/blob/cd853b625f41d8769eece8d725204b1e842689df/.github/workflows/build-dev.yaml#L101-L123